pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -666,6 +679,32 @@ func (h *Health) CheckClusterNodes(
 	return nil, nil
 }
 
+// CheckNodeAgentLeases checks if all given nodes have a corresponding lease
+func CheckNodeAgentLeases(nodeList *corev1.NodeList, leaseList *coordinationv1.LeaseList, clock clock.Clock) error {
+	if len(leaseList.Items) == 0 {
+		return fmt.Errorf("no leases")
+	}
+
```

Comment:
```suggestion
```

Remove this - otherwise, the migration scenario could fail (when `UseGardenerNodeAgent` is enabled but `gardener-node-agent` was not yet deployed because the `Shoot` wasn't reconciled yet).